### PR TITLE
Handle plan preview workflow's error and exit with error status code

### LINKED ISSUE #4661: Must handle plan preview workflow's error and exit with error status code
**What would you like to be added**:
Handle plan-preview workflow's error and exit with error status code.

**Why is this needed**:
Plan preview workflow comments error message but CI workflow passed.
It makes us very confused.

Screenshot_2023-11-08_at_16_40_17

### PR DESCRIPTION
**Pull request by TakumaKurosawa**
**What this PR does / why we need it**:
Plan preview workflow comments error message but CI workflow passed.
It makes us very confused.

**Which issue(s) this PR fixes**:

Fixes #4661 

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
  - Plan preview CI workflow will failed when result has error
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: tool/actions-plan-preview/main.go [hunk 1]
```diff
@@ -139,6 +139,9 @@ func main() {
 			doComment(failureBadgeURL + "\nUnable to run plan-preview for a closed pull request.")
 			return
 		}
+		body := makeCommentBody(event, result)
+		doComment(failureBadgeURL + body)
```
[the remark this answers is not in the document]
Comment (TakumaKurosawa) [reply tool/actions-plan-preview/main.go]: Exactly 👍 
I applied!!
https://github.com/pipe-cd/pipecd/pull/4662/commits/7343e29780822e629b44643e73e2ad9c290754a0

File: tool/actions-plan-preview/main.go [hunk 2]
```diff
@@ -139,6 +139,9 @@ func main() {
 			doComment(failureBadgeURL + "\nUnable to run plan-preview for a closed pull request.")
 			return
 		}
+		body := makeCommentBody(event, result)
+		doComment(failureBadgeURL + "\n" + body)
+		log.Fatal("plan-preview result has error")
```
Comment (kurochan) [inline tool/actions-plan-preview/main.go:144]: This case is an expected error. Therefore, it is inappropriate to set the log level to fatal.
If you just want to call exit(1), I think you can simply call exit(1).

Comment (TakumaKurosawa) [reply tool/actions-plan-preview/main.go:144]: @kurochan 
You mean fatal log level must be used when unexpected error, right?
I agree with your point 👍
I'll fix later, thanks.

Comment (TakumaKurosawa) [reply tool/actions-plan-preview/main.go]: @kurochan 
Applied!!
https://github.com/pipe-cd/pipecd/pull/4662/commits/23a79cdea1772ebe9512a799ecd87c872d4604d3

**Review by khanhtc1202 — APPROVED** [review]
Here you go 🚀

**Pull request merged**